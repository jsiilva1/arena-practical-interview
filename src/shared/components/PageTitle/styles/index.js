import styled from 'styled-components';

import { device } from '../../../theme/settings/screens';

export const Heading = styled.div`
  width: 100%;

  @media ${device.mobile} {
    margin: 0 auto;
    
    text-align: center;
    font-size: 1rem;
  }
`;