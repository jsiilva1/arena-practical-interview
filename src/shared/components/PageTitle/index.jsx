/**
 * External Dependencies
 */
import React from 'react';

/**
 * Internal Dependencies
 */
import { Heading } from './styles';

/**
 * Component Page Container
 *
 * Render a default container to content of pages
 * @return {*}
 */
const PageTitle = (props) => (
  <Heading>
    <h1>{props.text}</h1>
  </Heading>
);

export default PageTitle;
