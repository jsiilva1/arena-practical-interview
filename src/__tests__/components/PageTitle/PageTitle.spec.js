import React from 'react';
import { shallow } from 'enzyme';

import PageTitle from '../../../shared/components/PageTitle';

describe('PageTitle Component', () => {
  it('should render a h1 with a title passed by props', () => {
    const wrapper = shallow(
      <PageTitle text="Arena Platform" />
    );
    
    expect(wrapper).toMatchSnapshot();
  });
});
