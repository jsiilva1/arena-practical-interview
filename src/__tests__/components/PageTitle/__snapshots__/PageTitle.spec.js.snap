// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle Component should render a h1 with a title passed by props 1`] = `
<styled.div>
  <h1>
    Arena Platform
  </h1>
</styled.div>
`;
